src/Control/Carrier/Simple.hs

```diff
@@ -0,0 +1,115 @@
+
+2. Simple effect types cannot contain effectful sub-expressions as arguments. This precludes /scoped/ effects such as 'Control.Effect.Reader.local'. However, many useful effects, like 'Control.Effect.State.State' are not scoped, and thus representable as a first-order effect. (Note that the "Control.Carrier.Interpret" module provides an interface that allows for scoped effects, at the cost of a more complicated API.) Because effects that encapsulate some underlying 'State' effect are common, we provide the 'SimpleStateC' and 'interpretState' helpers, which wrap the strict state monad.
+
+Should your performance requirements change, it is generally possible to reformulate the interpretation of an effect with 'Algebra' without breaking client code.
+
+This module is based on (and almost entirely API-compatible with) work done at [FOSSA](https://fossa.com) and released as part of their [fossa-cli](https://github.com/fossas/fossa-cli) project. We are grateful for their contributions.
+
+@since 1.2.0.0
+
+= Defining effects with 'Simple'
+
+Effect types compatible with 'Simple' take fewer type parameters than 'Algebra'-compatible types. Here is the classic teletype example. Note that we define a type synonym for @Simple TeletypeF@ so that the implementation details of this effect do not leak into clients.
+
+@
+  data TeletypeF a where
+    Read :: TeletypeF String
+    Write :: String -> TeletypeF ()
+
+  type Teletype = Simple TeletypeF
+@
+
+Whereas an 'Algebra'-compatible @TeletypeF@ effect would take an @m@ monadic parameter and @k@ continuation parameter, those are elided in the simple formulation, as this formulation is intended to hide the details of monads and their associated state parameters.
+
+Effects are constructed by wrapping these effect types with 'sendSimple':
+
+@
+  read :: Has Teletype sig m => m String
+  read = sendSimple Read
+
+  write :: Has Teletype sig m => String -> m ()
+  write = sendSimple . Write
+@
+
+Defining an interpretation function is easy:
+
+@
+  runTeletypeIO :: MonadIO m => TeletypeF a -> m a
+  runTeletypeIO t = liftIO $ case t of
+    Read -> getLine
+    Write s -> putStrLn s
+@
+
+-}
+
+module Control.Carrier.Simple
```

Comment:
This should probably be `Control.Carrier.Interpret.Simple`, and `SimpleC` should be `InterpretC`.

Reply:
Also, `@since` annotations for new declarations would be greatly appreciated 🙇🏻‍♂️